cabal-install/Distribution/Client/ProjectConfig/Types.hs

```diff
@@ -309,7 +309,8 @@ data SolverSettings
        solverSettingLocalRepos        :: [FilePath],
        solverSettingConstraints       :: [(UserConstraint, ConstraintSource)],
        solverSettingPreferences       :: [Dependency],
-       solverSettingFlagAssignment    :: FlagAssignment, --TODO: [required eventually] must be per-package, not global
+       solverSettingFlagAssignment    :: FlagAssignment, --TODO: [required eventually] eliminate this global one
```

Comment:
Thanks, yes I have changed my mind on this one. It's ok to have a shared one I think, though the handling thereof could be improved. I'll adjust the comments.